### PR TITLE
KT-15537: Add inspection + intention to replace IntRange.start/endInclusive with first/last

### PR DESCRIPTION
**Pull request by devbridie**
This inspection reports usages of IntRange start and endInclusive. These properties can be replaced with equivalent `kotlin.collections` `first` and `last`.

Example: range.start` can be replaced with range.first.

Issue: https://youtrack.jetbrains.com/issue/KT-15537

The issue explicitly mentions `IntRange`, so this has been done, but I would imagine that it should work on all ranges `ClosedRange<T>` which also implement `Iterable<T>` e.g. `LongRange`, `UIntRange`, `CharRange`, etc. Thoughts?

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceIntRangeStartEndInclusiveWithFirstLastInspection.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                    ReplaceIntRangeEndInclusiveWithLastQuickFix()
+                )
+            }
+        }
+    }
+}
+
+private fun KtDotQualifiedExpression.getFunctionName(): FqName? {
+    val context = this.analyze()
+    return this.getResolvedCall(context)?.resultingDescriptor?.fqNameOrNull()
```
Comment (mglukhikh) [inline idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceIntRangeStartEndInclusiveWithFirstLastInspection.kt]: Here you can use `resolveToCall` which does not requires explicit context. It's a bit shorter.

File: idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceIntRangeStartEndInclusiveWithFirstLastInspection.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+private fun FqName.isStart(): Boolean {
+    return this == FqName("kotlin.ranges.IntRange.start")
```
Comment (mglukhikh) [inline idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceIntRangeStartEndInclusiveWithFirstLastInspection.kt]: I think it's a good idea to check here different range types, not only `IntRange`, and change inspection description accordingly.

**Comment by mglukhikh** [comment]
Sorry for late review.

**Comment by devbridie** [comment]
Rebased to current master and added support for other `Range` types.

**Comment by mglukhikh** [comment]
Merged manually, thank you.

**Comment by ilya-g** [comment]
Does the inspection explain _why_ these usages should be replaced?

**Pull request closed without merging**